### PR TITLE
Dbudzins/inversify classes

### PR DESCRIPTION
**Pull request by budzworthy**
## Description

This refactor implements the classes as identifiers for the ioc container

### Steps completed:

According to our definition of done, I have completed the following steps:

- [ ] Acceptance criteria met
- [ ] Unit tests added
- [ ] Docs updated (including config and env variables)
- [ ] Translations added
- [ ] UX tested
- [ ] Browsers / platforms tested
- [ ] Rebased & ready to merge without conflicts
- [ ] Reviewed own code

### CODE REVIEW

File: src/stores/ProfileController.ts [hunk 1]
```diff
@@ -1,38 +1,42 @@
...
+    return this.profileService?.enterProfile({ id, pin }, this.getSandbox());
+  };
+
+  deleteProfile = ({ id }: ProfileDetailsPayload) => {
+    return this.profileService?.deleteProfile({ id }, this.getSandbox());
+  };
+
+  getProfileDetails = ({ id }: ProfileDetailsPayload) => {
+    return this.profileService?.getProfileDetails({ id }, this.getSandbox());
+  };
```
Comment (ChristiaanScheermeijer) [inline src/stores/ProfileController.ts:37]: Won't this result in the return type being a union `undefined | Promise<Result>`?

File: src/stores/ProfileController.ts [hunk 2]
```diff
@@ -1,38 +1,42 @@
...
+import { ProfileService } from '#src/services/profile.service';
+import { useConfigStore } from '#src/stores/ConfigStore';
+
+@injectable()
+export default class ProfileController {
+  private readonly profileService: ProfileService;
+
+  constructor(profileService: ProfileService) {
+    this.profileService = profileService;
+  }
```
Comment (ChristiaanScheermeijer) [inline src/stores/ProfileController.ts:13]: I guess the ProfileService isn't a concrete injection? 

https://github.com/inversify/InversifyJS/blob/master/wiki/constructor_injection.md

File: src/stores/ProfileController.ts (hunk 1 above)
Comment (budzworthy) [reply src/stores/ProfileController.ts:37]: Yes, that's very much intentional. Downstream code needs to understand that there might not be a profile service / profile controller / profile data. This is true elsewhere for things like checkout too.

File: src/stores/ProfileController.ts (hunk 2 above)
Comment (budzworthy) [reply src/stores/ProfileController.ts:13]: Is there a bug here? Maybe it needs to be optional (`?`)?It's possible, as the profile service was added after the fact and only came in when I merged latest from dev. I tried to refactor it quickly, but I might have missed something.

File: src/stores/ProfileController.ts (hunk 1 above)
Comment (ChristiaanScheermeijer) [reply src/stores/ProfileController.ts:37]: Ah I understand. So this can replace the feature flags 😄

File: src/stores/ProfileController.ts (hunk 2 above)
Comment (ChristiaanScheermeijer) [reply src/stores/ProfileController.ts:13]: No, I think it is correct like this. If the service was concrete, I would prefer using the concrete injection method so we can remove all optional chaining. But your other comment also answered this question 😄

**Pull request closed without merging**